opentelemetry-user-events-logs/src/logs/exporter.rs

```diff
@@ -51,59 +73,36 @@ impl UserEventsExporter {
-        );
-        eventheader_provider.register_set(
-            eventheader::Level::Warning,
-            exporter_config.get_log_event_keyword(),
-        );
-        eventheader_provider.register_set(
-            eventheader::Level::Error,
-            exporter_config.get_log_event_keyword(),
-        );
-        eventheader_provider.register_set(
-            eventheader::Level::CriticalError,
-            exporter_config.get_log_event_keyword(),
-        );
-
-        eventheader_provider.create_unregistered(
-            true,
-            eventheader::Level::Informational,
-            exporter_config.get_log_event_keyword(),
-        );
-        eventheader_provider.create_unregistered(
-            true,
-            eventheader::Level::Verbose,
-            exporter_config.get_log_event_keyword(),
-        );
-        eventheader_provider.create_unregistered(
-            true,
-            eventheader::Level::Warning,
-            exporter_config.get_log_event_keyword(),
-        );
-        eventheader_provider.create_unregistered(
-            true,
-            eventheader::Level::Error,
-            exporter_config.get_log_event_keyword(),
-        );
-        eventheader_provider.create_unregistered(
-            true,
-            eventheader::Level::CriticalError,
-            exporter_config.get_log_event_keyword(),
-        );
-
+        if exporter_config.keywords_map.len() == 0 {
+            println!(
+                "Register default keyworkd {}",
+                exporter_config.default_keyword
+            );
+            Self::register_events(&mut eventheader_provider, exporter_config.default_keyword)
+        }
+        for keyword in exporter_config.keywords_map.values().into_iter() {
+            Self::register_events(&mut eventheader_provider, *keyword)
+        }
         UserEventsExporter {
             provider: Arc::new(eventheader_provider),
             exporter_config,
         }
     }
 
+    fn register_events(eventheader_provider: &mut eventheader_dynamic::Provider, keyword: u64) {
+        eventheader_provider.register_set(eventheader::Level::Informational, keyword);
+        eventheader_provider.register_set(eventheader::Level::Verbose, keyword);
+        eventheader_provider.register_set(eventheader::Level::Warning, keyword);
+        eventheader_provider.register_set(eventheader::Level::Error, keyword);
+        eventheader_provider.register_set(eventheader::Level::CriticalError, keyword);
+
+        eventheader_provider.create_unregistered(true, eventheader::Level::Informational, keyword);
+        eventheader_provider.create_unregistered(true, eventheader::Level::Verbose, keyword);
+        eventheader_provider.create_unregistered(true, eventheader::Level::Warning, keyword);
+        eventheader_provider.create_unregistered(true, eventheader::Level::Error, keyword);
+        eventheader_provider.create_unregistered(true, eventheader::Level::CriticalError, keyword);
```

Comment:
What is the `create_unregistered` for?

`create_unregistered` is a no-op if the event set is already registered, which it should be because you just called `register_set`.

`create_unregistered` is "for testing purposes".

Reply:
Thanks for clarifying. Do we need to do anything explicit to ensure that the events are unregistered (i,e tracepoint names are removed) once the application shutdowns?

Reply:
The event will be unregistered when the provider is dropped.

Reply:
Thanks, have removed `create_unregistered` calls here.